pkg/apis/uiplugin/v1alpha1/types.go

```diff
@@ -31,20 +31,79 @@ type UIPluginList struct {
 	Items           []UIPlugin `json:"items"`
 }
 
-// +kubebuilder:validation:Enum=Dashboards
+// +kubebuilder:validation:Enum=Dashboards;TroubleshootingPanel;DistributedTracing
 type UIPluginType string
 
 const (
 	// TypeDashboards deploys the Dashboards Dynamic Plugin for OpenShift Console.
 	TypeDashboards UIPluginType = "Dashboards"
+	// DistributedTracing deploys the Distributed Tracing Dynamic Plugin for the OpenShift Console
+	TypeDistributedTracing UIPluginType = "DistributedTracing"
+	// TroubleshootingPanel deploys the Troubleshooting Panel Dynamic Plugin for the OpenShift Console
+	TypeTroubleshootingPanel UIPluginType = "TroubleshootingPanel"
 )
 
+// TroubleshootingPanelConfig contains options for configuring the Troubleshooting Panel  plugin
+type TroubleshootingPanelConfig struct {
+	// Timeout is the maximum duration before a query timeout.
+	//
+	// The value is expected to be a sequence of digits followed by a unit suffix, which can be 's' (seconds)
+	// or 'm' (minutes).
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([sm]{1})$"
+	Timeout string `json:"timeout,omitempty"`
+	// korrel8r defines the Korrel8r instance that the troubleshooting panel plugin will connect to
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance"
+	Korrel8r TroubleshootingPanelKorrel8rConfig `json:"korrel8r,omitempty"`
+}
+
+type TroubleshootingPanelKorrel8rConfig struct {
+	// Name of the korrel8r instance
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance Name"
+	Name string `json:"name,omitempty"`
+
+	// Namespace of the korrel8r instance
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance Namespace"
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// DistributedTracingConfig contains options for configuring the Distributed Tracing plugin
+type DistributedTracingConfig struct {
+	// Timeout is the maximum duration before a query timeout.
+	//
+	// The value is expected to be a sequence of digits followed by a unit suffix, which can be 's' (seconds)
+	// or 'm' (minutes).
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([sm]{1})$"
+	Timeout string `json:"timeout,omitempty"`
+}
```

Comment:
How is the distributed tracing panel connecting to a TempoStack instance? Do we need something like @xperimental created for the logging-ui plugin [here](https://github.com/rhobs/observability-operator/pull/477/files#diff-16e5f13859d1d0daf88ef24d58309d2a62d1768a556ca142b9482722e50a68d5R67-R69), i.e:

```go
// DistributedTracingConfig contains options for configuring the Distributed Tracing plugin
type DistributedTracingConfig struct {
	// TempoStack points to the TempoStack instance of which logs should be displayed.
	//
	// +kubebuilder:validation:Required
	TempoStack TempoStackReference `json:"tempoStack"`

	// Timeout is the maximum duration before a query timeout.
	//
	// The value is expected to be a sequence of digits followed by a unit suffix, which can be 's' (seconds)
	// or 'm' (minutes).
	//
	// +optional
	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
	// +kubebuilder:validation:Pattern:="^([0-9]+)([sm]{1})$"
	Timeout string `json:"timeout,omitempty"`
}


// TempoStackReference is used to configure a reference to a TempoStack that should be used
// by the Distributed Tracing console plugin.
//
// +structType=atomic
type TempoStackReference struct {
	// Name of the TempoStack resource.
	//
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MinLength:=1
	Name string `json:"name"`

	// Namespace of the TempoStack resource.
	//
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MinLength:=1
	Namespace string `json:"namespace"`
}
```

Reply:
There is no single TempoStack instance. We [list](https://github.com/openshift/distributed-tracing-console-plugin/blob/main/pkg/server.go#L96) the available TempoStack instances so the frontend can choose one to connect.

Reply:
Ah good to know, didn't have that in my mind when reviewing the code.